src/server/ApiRoutes/Hold.js

```diff
@@ -69,6 +72,31 @@ function postHoldAPI(
 }
 
 /**
+ * mapLocationDetails(locations)
+ * The function extracts the details of the delivery locations from the location.js and
+ * locationCodes.js based on the locastion id we get from deliveryLocationsByBarcode API.
+ *
+ * @param {array} locations
+ * @return {array}
+ */
+function mapLocationDetails(locations) {
+  locations.map(loc => {
+    _mapObject(locationCodes, (c) => {
```

Comment:
Yeah that's what I have learned when I see the data sheet. I think at most there'll be only 5 addresses as to 5 research branches. But I am not sure.